File: AwesomeArchComponent/app/src/main/java/com/akexorcist/repositoryarchcomponents/repo/hero/HeroRepository.java
package com.akexorcist.repositoryarchcomponents.repo.hero;

import android.arch.lifecycle.LiveData;

import com.akexorcist.repositoryarchcomponents.api.Resource;
import com.akexorcist.repositoryarchcomponents.api.hero.HeroService;
import com.akexorcist.repositoryarchcomponents.api.hero.response.HeroResult;
import com.akexorcist.repositoryarchcomponents.db.hero.HeroDao;
import com.akexorcist.repositoryarchcomponents.repo.AppExecutors;

/**
 * Created by Akexorcist on 10/6/2017 AD.
 */

public class HeroRepository {
    private static HeroRepository heroRepository;

    public static HeroRepository getInstance() {
        if (heroRepository == null) {
            heroRepository = new HeroRepository();
        }
        return heroRepository;
    }

    private HeroService heroService;
    private HeroDao heroDao;
    private AppExecutors appExecutors;

    private HeroRepository() {
        heroService = null;
        heroDao = null;
        appExecutors = null;
    }

    public LiveData<Resource<HeroResult>> getHero(Boolean isForceFetch) {
        return null;
    }

    public void clearAllHeroesInDatabase() {

    }
}

File: AwesomeArchComponent/app/src/main/java/com/akexorcist/repositoryarchcomponents/repo/hero/HeroRepository.kt
package com.akexorcist.repositoryarchcomponents.repo.hero

import android.arch.lifecycle.LiveData

import com.akexorcist.repositoryarchcomponents.api.Resource
import com.akexorcist.repositoryarchcomponents.api.hero.HeroService
import com.akexorcist.repositoryarchcomponents.api.hero.respond.HeroResult
import com.akexorcist.repositoryarchcomponents.hero.HeroDao
import com.akexorcist.repositoryarchcomponents.repo.AppExecutors

/**
 * Created by Akexorcist on 10/6/2017 AD.
 */

class HeroRepository private constructor() {

    private val heroService: HeroService?
    private val heroDao: HeroDao?
    private val appExecutors: AppExecutors?

    init {
        heroService = null
        heroDao = null
        appExecutors = null
    }

    fun getHero(isForceFetch: Boolean?): LiveData<Resource<HeroResult>>? {
        return null
    }

    fun clearAllHeroesInDatabase() {

    }

    companion object {
        private var heroRepository: HeroRepository? = null

        val instance: HeroRepository
            get() {
                if (heroRepository == null) {
                    heroRepository = HeroRepository()
                }
                return heroRepository as HeroRepository
            }
    }
}
